### PR TITLE
API assigns points to each Fika

### PR DESCRIPTION
**Pull request by dorianbuck**
PT: https://www.pivotaltracker.com/story/show/180227873

## User Story
```
As an API
In order to encourage participation in the system
I would like to assign points to each participant
```

## Tasks
- Create a score service
- Add attr to Fikas table 
- Add attr to participants table
- Update serializers to return fikaScore
- Assign more points to the participant who has been at the company the shortest
- - Always 10 points per fika
- - When the difference in start dates is < 5 years points are split  [5, 5], <10 years = [6,4], > 10 years = [7,3]

### CODE REVIEW

File: spec/requests/api/fikas/create_spec.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@
     it { is_expected.to have_http_status 201 }
 
     it 'is expected to return a message' do
-      expect(response_json['message']).to eq 'Fikas successfully created'
+      expect(response_json['message']).to eq 'Fikas successfully created'
```
Comment (maxarvid) [inline spec/requests/api/fikas/create_spec.rb]: not needed

File: spec/requests/api/participants/create_spec.rb [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
     it { is_expected.to have_http_status 201 }
 
-    it 'is expected to return a message' do
+    it 'is expected to return a message' do
```
Comment (maxarvid) [inline spec/requests/api/participants/create_spec.rb]: not needed

File: app/models/fika.rb [hunk 3]
```diff
@@ -19,4 +20,15 @@ def create_calendar_entry
   rescue StandardError
     raise StandardError, 'We are experiencing problems with Google calendar at the moment, please try again later!'
   end
+
+  def assign_participant_fika_scores
+    participant_1 = Fika.last.participants[0]
+    participant_2 = Fika.last.participants[1]
+    
+    participant_1.fika_score = participant_1.fika_score + Fika.last.score[0]
+    participant_2.fika_score = participant_2.fika_score + Fika.last.score[1]
```
Comment (maxarvid) [inline app/models/fika.rb]: Refactor to a +=

File: app/services/score_service.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    when 0..5 #years
+      [5, 5]
+    when 5..10 #years
+      is_positive(calc_date_difference) ? [6, 4] : [4, 6]
+    else
+      is_positive(calc_date_difference) ? [7, 3] : [3, 7]
+    end
+  end
+
+  def self.is_positive(num)
```
Comment (maxarvid) [inline app/services/score_service.rb]: `positive?` is built into Ruby, no need to have our own helper method.

File: app/services/score_service.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def self.difference_in_years(calc_date_difference)
+    (calc_date_difference / (60 * 60 * 24 * 365)).abs 
+  end
+
+  def self.assign_scores(difference_in_years, calc_date_difference)
+    case difference_in_years
+    when 0..5 #years
+      [5, 5]
```
Comment (maxarvid) [inline app/services/score_service.rb]: I suggest we extract these arrays to variables we can name at the top of this file. We could call them EVEN_POINT_DISTRIBUTION and so forth, then it's just a matter of inverting them in the ternaries below.

File: spec/services/assign_fika_score_spec.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
+RSpec.describe 'AssignScoreService', type: :request do
+  let!(:participant_1) { create(:participant, start_date: Time.now, fika_score: 0) }
+  let(:admin) { create(:admin) }
+  let(:credentials) { admin.create_new_auth_token }
+
+  describe 'when the participants have a delta between their start dates of under 5 years' do
```
Comment (maxarvid) [inline spec/services/assign_fika_score_spec.rb]: I never thought I would say this, but maybe this language is a little too mathsy if we want our specs to serve as documentation? I suggest changing `delta` here to `difference`.

File: spec/services/assign_fika_score_spec.rb [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    it 'is expected that the fika score will be split evenly between the participants' do
+      expect(Fika.last.score).to eq([5, 5])
+    end
+
+    it 'is expected that participant 1 will have a fika score of 5' do
+      expect(Participant.last.fika_score).to eq 5
+    end
+  end
+
+  describe 'when the participants have a delta between their start dates of over 5 years' do
```
Comment (maxarvid) [inline spec/services/assign_fika_score_spec.rb]: As above

File: spec/services/assign_fika_score_spec.rb [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    it 'is expected that the fika score will not be split evenly' do
+      expect(Fika.last.score).to eq([6, 4])
+    end
+
+    it 'is expected that participant 1 will have a fika score of 6' do
+      expect(Participant.first.fika_score).to eq 6
+    end
+  end
+
+  describe 'when the participants have a delta between their start dates of over 10 years' do
```
Comment (maxarvid) [inline spec/services/assign_fika_score_spec.rb]: as above

File: app/models/fika.rb (hunk 3 above)
Comment (Fraseer) [reply app/models/fika.rb]: Cool didnt know you could do that

**Review by AugustEnhager — APPROVED** [review]

**Pull request merged**